
<div class="container-xl">
  <div class="sidenav">
      <a (click)="scroll(singleseq)"><h5><b>Sequence</b></h5></a>
      <a (click)="scroll(indexesloc)"><h6>Index location</h6></a>
      <a (click)="scroll(wrapline)"><h6>Wrap line</h6></a>
      <a (click)="scroll(iconsblock)"><h6>Icons</h6></a>
      <a (click)="scroll(regionsblock)"><h6> Regions</h6></a>
      <a (click)="scroll(patternsblock)"><h6>Patterns</h6></a>

      <a (click)="scroll(msablock)"><h5><b>MSA</b></h5></a>
      <a (click)="scroll(color)"><h6>Color</h6></a>
      <a (click)="scroll(colormatrix)"><h6>Color matrix</h6></a>
      <a (click)="scroll(consensus)"><h6>Consensus</h6></a>
  </div>

  <div class="row">
<!--    <div class="col-2">-->

<!--    </div>-->

    <div class="col">

      <h1 class="txt">Getting Started</h1>
      <p class="mb-5">
        ProSeqViewer provides both single sequence and multiple sequence alignment (MSA) visualization.
        Check our Getting started guide to get familiar with the library. To test custom values immediately without
        installing the tool try our <a href="https://jsfiddle.net/rvqgch61/" target="_blank"><b>jsFiddle</b></a>
      </p>


      <h2 #singleseq class="txt">Single sequence</h2>

      <h6 #indexesloc class="snippet-title">Index location</h6>
      <p>
        Indexes refer to sequence positions and can be provided by the user or automatically calculated by the library.
        Indexes are disabled by default, you can activate them and choose indexes location, lateral or on top.
      </p>

      <p style="color:#000;"><b>Lateral index</b></p>
      <div id="sqv5" class="sqvDemo"></div>
      <pre class="codebkg">{{latindexoptions | json}}</pre>

      <p style="color:#000;"><b>Top index</b></p>
      <div class="sqvDemo"><div id="sqv10"></div></div>
      <pre class="codebkg">{{topindexoptions | json}}</pre>



      <!-------------------------------------  wrapline  EXAMPLE ---------------------------------------->

      <h6 #wrapline class="snippet-title">Wrap Line</h6>
      <p>
        The sequence can be broken into chunks of fixed length to improve readability.
        Chunk length is controlled by the <strong>chunkSize</strong> option.
        By default, the viewer is responsive and blocks are automatically moved to a new line when they
        exceed the container width. Sometimes, e.g. for long sequences, it is preferable to see
        the sequence without new line breaks. Sequence wrapping is controlled by <b>wrapLine</b>
      </p>

      <p style="color:#000;"><b>Responsive (default)</b></p>

      <div class="sqvDemo">
        <div class="row">
          <div id="sqv8"></div>
        </div>
      </div>

      <pre class="codebkg">{{wrapoptionsnr2 | json}}</pre>

      <p style="color:#000;"><b>Non-responsive</b></p>

      <div class="sqvDemo">
        <div class="row">
          <div id="sqv13"></div>
        </div>
      </div>

      <pre class="codebkg">{{wrapoptionsnr1 | json}}</pre>


      <!---------------------------------------------- Icons -------------------------------------------------->
      <h6 #iconsblock class="snippet-title">Icons</h6>
      <p>
        Icons can be added on top of the sequence. The library provides some predefined icons:
        <strong>helix, turn, lollipop, strand, arrowLeft, arrowRight, noSecondary</strong>. Alternatively custom svg icons can be provided, see <a routerLink="/doc"> Documentation</a>
      </p>


      <div class="sqvDemo" id="sqvSingle"></div>
      <p>
        Secondary structure of the human ASL protein (UniProt P04424, <i>Homo sapiens</i>)
      </p>
      <pre class="codebkg">{{icons | json}}</pre>



      <!--------------------------------------------  REGIONS  EXAMPLE ---------------------------------------->

      <h6 #regionsblock class="snippet-title">Regions</h6>
      <div>Highlight part of the sequence with customizable colors</div>

      <div class="sqvDemo"  id="sqv2"></div>
      <p>Structurally repeated units (blu, red) of the Tryptophan synthase alpha chain
        (PDB 6UB9 chain A, <i>Mycobacterium tuberculosis</i>). Insertions are in yellow</p>

      <pre class="codebkg">{{regions | json}}</pre>




      <!-------------------------------------  PATTERNS  EXAMPLE ---------------------------------------->

      <h6 #patternsblock class="snippet-title">Patterns</h6>
      <p>Match the sequences against a regular expression to show possible patterns. </p>
      <div class="sqvDemo" id="sqv4"></div>

      <pre class="codebkg">{{patterns | json}}</pre>



      <!--------------------------------------------  MSA  section ---------------------------------------->
      <h2 #msablock class="txt"> Multiple sequence alignment (MSA)</h2>



      <div class="sqvDemo" id="sqv0"></div>
      <p>
        Multiple sequence alignment of the Aspartate ammonia-lyase protein (UniProt P0AC38) with ten homologs. Residue numbering displayed on top of the sequences
      </p>
      <pre class="codebkg">{{msa | json}}</pre>



      <!-------------------------------------  sequenceColor clustal  EXAMPLE ---------------------------------------->

      <h6 #color class="snippet-title">Color</h6>
      <p>
        Amino acids can be colored based on predefined or custom color schemes. Predefined color sets:
        <strong>clustal, zappo, taylor, hydrophobicity</strong>, are defined as in JalView, see
        <a href="https://www.jalview.org/help/html/colourSchemes/index.html">here</a>.
        Check the <a routerLink="/doc">documentation</a> of the <strong>sequenceColor</strong> option parameter for custom coloring
      </p>

      <p>
        <b>Clustal coloring</b>
      </p>

      <div class="circlez q">N, Q, S, P, T, H, Y</div>
      <div class="circlez a">A, I, L, M, F, W, V </div>
      <div class="circlez k">K, R</div><div class="circlez e">E, D</div>
      <div class="circlez c">C</div>

      <div class="sqvDemo" id="sqv6"></div>

      <pre class="codebkg">{{clustaloptions | json}}</pre>


      <!-------------------------------------  sequenceColorMatrix blosum  EXAMPLE ---------------------------------------->
      <h6 #colormatrix class="snippet-title">Color matrix</h6>
      <p>
        Amino acids can be colored based on a comparison with the first sequence (the MSA seed).
        It is possible to provide a custom matrix representing all possible amino acids combinations and a color.
        ProSeqViewer provides a predefined BLOSUM62 matrix that includes predefined colors
        proportional to the corresponding substitution values (colors for negative values are not assigned)
      </p>

      <p>
        <b>Blosum matrix</b>
      </p>
      <div class="sqvDemo" id="sqvBlosum"></div>
      <pre class="codebkg">{{blosumdefaultoptions | json}}</pre>

      <img src="../../assets/heatmap.svg" class="figure figure-img ounded mx-auto d-block">

      <p class="text-center">
        Predefined colors in the BLOSUM62 matrix, negative values are set to zero (white). The diagonal represent the frequency of that amino acid in representative organisms
      </p>


      <p><b>Custom matrix</b></p>
      <p>
        Custom color matrix can be provided with the <b>sequenceColorMatrix</b> field</p>

      <div class="sqvDemo" id="sqvBlosumCustom"></div>
      <pre class="codebkg">{{blosumcustomoptions | json}}</pre>




      <!-------------------------------------  CONSENSUS  EXAMPLE ---------------------------------------->

      <h6 #consensus class="snippet-title">Consensus</h6>
      <p>
        ProSeqViewer provides two types of consensus, <strong>identity</strong> and <strong>physical</strong>. The consensus is
        calculated based on amino acid conservation (identity or physical property) over MSA columns and it is added as a new sequence on top.
        <br>
        The most frequent amino acid (or property) is reported.
        When the conservation is below a given threshold (<strong>consensusDotThreshold</strong> option, default 90%) the consensus displays a <strong>dot (.) symbol</strong>.
        <br>
        It is possible to set a custom consensus with personalized colors, thresholds and symbols
      </p>

      <!--  identity -->
      <div class="palette">
        <p>
          <b>Identity</b>
        </p>

        <p>
          Different conservation ranges are colored based on predefined colors
        </p>

        <div class="consensusrect aa">&nbsp;100%</div>
        <div class="consensusrect bb">&nbsp;70%</div>
        <div class="consensusrect cc">&nbsp;40%</div>
        <div class="consensusrect dd">&nbsp;10%</div>
      </div>

      <div class="sqvDemo"><div id="sqv7"></div></div>


      <pre class="codebkg">{{consensusidentity | json}}</pre>


      <!--  physical-->
      <p>
        <b>Physical</b>
      </p>
      <p>
        The most conserved physico-chemical property is reported and colored according to a predefined scheme. If the conservation
        is 100% the amino acid is reported
      </p>

      <div class="badge-feature me-1" style="background-color: black;">X</div>Conserved (100% identity) <span class="me-3"></span>
      <div class="badge-feature me-1" style="background-color: #f09048;">n</div> small nonpolar (A,G,S,T) <span class="me-3"></span>
      <div class="badge-feature me-1" style="background-color: #558B6E;">h</div> small hydrophobic (C,V,I,L,P,F,Y,M,W ) <span class="me-3"></span>
      <div class="badge-feature me-1" style="background-color: #A10702;">~</div> negatively charged (D,E) <span class="me-3"></span>
      <div class="badge-feature me-1" style="background-color: #3626A7;">+</div> positively charged (K,R)


      <div class="sqvDemo" id="sqvPhysical"></div>

      <pre class="codebkg">{{consensusphysical | json}}</pre>


    </div>

  </div>

</div>
